src/catchup/ApplyLedgerWork.cpp

```diff
@@ -23,7 +24,15 @@ BasicWork::State
 ApplyLedgerWork::onRun()
 {
     ZoneScoped;
-    mApp.getLedgerManager().closeLedger(mLedgerCloseData);
+    try
+    {
+        mApp.getLedgerManager().closeLedger(mLedgerCloseData);
+    }
+    catch (std::exception const& e)
```

Comment:
I am not quite sure this is the right fix:
with this change, we stop crashing if there is some catastrophic problem when closing a ledger, and just retry etc

the reason we want to crash is that the core file may contain critical information that would allow to diagnose the issue.

I *think* that the original problem is not so much that we crash but that we crash with the wrong context (an assert in Work).

Reply:
Looking at the kind of exceptions that `CloseLedger` can throw, I am not sure there are any situations where we want to fail the "work" as all situations indicate some severe problem with the local node
(I was trying to see if we should crash only for a subset of exceptions, like what was done before in `ApplyCheckpointWork`)